cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -278,18 +308,17 @@ private void addProcessingStats(List<MetricValues> records, long currentTime) {
                                           new MetricValue("metrics.process.delay.ms", MetricType.GAUGE, delay))));
     }
 
-    private void persistMessageId(byte[] messageId) {
+    private void persistMessageIds(Map<TopicIdMetaKey, byte[]> metaKeyMap) {
```

Comment:
on line#298, can you also log the current time in the end. so user checking metrics.processor logs can get an idea immediately about the delay.